components/Layout/Header.tsx

```diff
@@ -0,0 +1,107 @@
+import Link from 'next/link';
+import { useState } from 'react';
+import { hoverClass, headerLinks } from 'styles/globals';
+
+export const Header = () => {
+  const [showMobileMenu, setShowMobileMenu] = useState(false);
+  const toggleMobileMenu = () => {
+    setShowMobileMenu(!showMobileMenu);
```

Comment:
can we do this instead, this approach is more reliable https://www.reddit.com/r/reactjs/comments/z6zl3x/setstateprevious_previous_1_or_setstatecurrent_1/?rdt=61444

```suggestion
    setShowMobileMenu(prev => !prev);
```